src/messaging/tests/TestReliableMessageProtocol.cpp

```diff
@@ -1443,6 +1443,78 @@ void CheckLostStandaloneAck(nlTestSuite * inSuite, void * inContext)
     NL_TEST_ASSERT(inSuite, rm->TestGetCountRetransTable() == 0);
 }
 
+struct BackoffComplianceTestVector
+{
+    uint8_t sendCount;
+    System::Clock::Timestamp backoffBase;
+    System::Clock::Timestamp backoffMin;
+    System::Clock::Timestamp backoffMax;
+};
+
+struct BackoffComplianceTestVector theBackoffComplianceTestVector[] = {
+    {
+        .sendCount   = 0,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(300),
+        .backoffMax  = System::Clock::Timestamp(375),
+    },
+    {
+        .sendCount   = 1,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(300),
+        .backoffMax  = System::Clock::Timestamp(375),
+    },
+    {
+        .sendCount   = 2,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(480),
+        .backoffMax  = System::Clock::Timestamp(600),
+    },
+    {
+        .sendCount   = 3,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(768),
+        .backoffMax  = System::Clock::Timestamp(960),
+    },
+    {
+        .sendCount   = 4,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(1229),
+        .backoffMax  = System::Clock::Timestamp(1536),
+    },
+    {
+        .sendCount   = 5,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(1966),
+        .backoffMax  = System::Clock::Timestamp(2458),
+    },
+    {
+        .sendCount   = 6,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(1966),
+        .backoffMax  = System::Clock::Timestamp(2458),
+    },
+};
+
+const unsigned theBackoffComplianceTestVectorLength =
+    sizeof(theBackoffComplianceTestVector) / sizeof(struct BackoffComplianceTestVector);
+
+void CheckGetBackoff(nlTestSuite * inSuite, void * inContext)
+{
+    TestContext & ctx       = *reinterpret_cast<TestContext *>(inContext);
+    ReliableMessageMgr * rm = ctx.GetExchangeManager().GetReliableMessageMgr();
```

Comment:
Please make the method static or a free standing function, so you do not need to access it via a ReliableMessageManager object in tests.